sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/xmlExpressions.scala

```diff
@@ -186,9 +186,6 @@ case class SchemaOfXml(
   @transient
   private lazy val xmlOptions = new XmlOptions(options, "UTC")
 
-  @transient
-  private lazy val xmlFactory = xmlOptions.buildXmlFactory()
```

Comment:
This variable `xmlFactory` is not used in the expression `SchemaOfXml`, so it is deleted here. It has no strong relationship with this PR.